substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/RuntimeSupport.java

```diff
@@ -42,8 +45,29 @@
 
 import jdk.graal.compiler.api.replacements.Fold;
 
-@AutomaticallyRegisteredImageSingleton({VMRuntimeSupport.class, RuntimeSupport.class})
-public final class RuntimeSupport implements VMRuntimeSupport {
+@AutomaticallyRegisteredImageSingleton({VMRuntimeSupport.class, RuntimeSupport.class, LibGraalRuntimeSupport.class})
+public final class RuntimeSupport implements VMRuntimeSupport, LibGraalRuntimeSupport {
+
+    @Override
```

Comment:
I think you should move all these new methods to a separate libgraal-specific image singleton that implements `LibGraalRuntimeSupport` (for a normal native image, it does not make sense to use any of these methods).

Reply:
Ok. Where would you suggest to put `LibGraalRuntimeSupportImpl`? In `substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/libgraal/LibGraalRuntimeSupportImpl.java`?

Reply:
Sounds good.

Reply:
Will do.